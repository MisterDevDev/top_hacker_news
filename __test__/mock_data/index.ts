import path from 'path'
import { promises as fs } from "fs";
import { IComment, IStory } from '@/typings';

const JSON_FILEPATH = path.join(__dirname, "json");

export async function readFakeData() {
    const [fakeStories, fakeIds, fakeComment] = await Promise.all([
        getJSONfromFile(filenames.stories),
        getJSONfromFile(filenames.storyIds),
        getJSONfromFile(filenames.comment),
    ])

    return {
        fakeStories: fakeStories as IStory[],
        fakeIds: fakeIds as number[],
        fakeComment:fakeComment as IComment
    }
}

enum filenames {
    stories = "stories.json",
    storyIds = "storyIds.json",
    comment = "comment.json"
  }

  type JsonDataType =
  | IStory[]
  | number[]
  | IComment

    async function getJSONfromFile<ItemType extends JsonDataType>(
    filename: filenames,
  ): Promise<ItemType> {
    const filePath = path.join(JSON_FILEPATH , filename);
    const data = await fs.readFile(filePath);
    return JSON.parse(data.toString());
  }