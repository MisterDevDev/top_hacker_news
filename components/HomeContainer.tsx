import { story } from "@/typings";

interface Props {
  topStories: story[];
}

export const HomeContainer = (props: Props) => {
  return (
    <main style={{ display: "flex", flexWrap: "wrap" }}>
      {props.topStories.map((story: story) => {
        return (
          <div key={story.id} style={{ margin: "1rem" }}>
            <ul style={{ overflow: "hidden" }}>
              <li>by: {story.by}</li>
              <li>{story.title}</li>
              <li>Comments: {story.descendants}</li>
              <li>Score: {story.score}</li>
              <li>{story.url}</li>
              <li>Time: {story.time}</li>
            </ul>
          </div>
        );
      })}
    </main>
  );
};
