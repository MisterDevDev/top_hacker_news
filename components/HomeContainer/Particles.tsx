import { useCallback } from "react";
import Particles from "react-tsparticles";
import type { Engine } from "tsparticles-engine";
import { loadFull } from "tsparticles";

export const HeroParticles = () => {
  const particlesInit = useCallback(async (engine: Engine) => {
    await loadFull(engine);
  }, []);

  const particlesLoaded = useCallback(async () => {}, []);

  return (
    <Particles
      id="tsparticles"
      init={particlesInit}
      loaded={particlesLoaded}
      options={{
        fpsLimit: 80,
        fullScreen: {
          enable: false,
          zIndex: 1,
        },
        interactivity: {
          events: {
            onClick: {
              enable: true,
              mode: "push",
            },
            onHover: {
              enable: true,
              mode: "repulse",
            },
            resize: true,
          },
          modes: {
            push: {
              quantity: 4,
            },
            repulse: {
              distance: 200,
              duration: 0.4,
            },
          },
        },
        particles: {
          color: {
            value: "#ffffff",
          },
          links: {
            color: "#ffffff",
            distance: 150,
            enable: true,
            opacity: 0.7,
            width: 1,
          },
          collisions: {
            enable: false,
          },
          move: {
            direction: "none",
            enable: true,
            bounce: false,
            random: false,
            speed: 5,
            straight: false,
          },
          number: {
            density: {
              enable: true,
              area: 800,
            },
            value: 80,
          },
          opacity: {
            value: 0.5,
          },
          shape: {
            type: "circle",
          },
          size: {
            value: { min: 1, max: 5 },
          },
        },
        detectRetina: true,
      }}
    />
  );
};
