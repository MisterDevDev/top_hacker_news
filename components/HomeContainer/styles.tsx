import styled from "styled-components";
import img from "../../public/hacker_news_hero.jpg";

export const StyledStoryContainer = styled.div`
  flex-direction: column;
  align-items: center;
  display: flex;
`;

export const StyledTitle = styled.h1`
  margin: 0;
  font-size: 8em;
  filter: brightness(100%);
  background-image: linear-gradient(
    ${({ theme }) => {
      return `to right, ${theme.CRIMSON} 15%, ${theme.DARK_SALMON} 25%, ${theme.CRIMSON} 55%, ${theme.DARK_SALMON} 100%`;
    }}
  );
  background-clip: text;
  -webkit-background-clip: text;
  color: transparent;
  @media only screen and (max-width: 1025px) {
    font-size: 5em;
  }
`;

export const StyledGradientText = styled.h4`
  position: absolute;
  top: 0;
  right: 25px;
  font-size: 3em;
  filter: brightness(100%);
  text-align: right;
  background-image: linear-gradient(
    ${({ theme }) => {
      return `to right, ${theme.DANGER} 15%, ${theme.TEXT_DANGER} 25%, ${theme.DANGER} 55%, ${theme.TEXT_DANGER} 100%`;
    }}
  );
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  @media only screen and (max-width: 1025px) {
    font-size: 2em;
  }
`;

export const StyledSubtitleBox = styled.div`
  position: relative;
  padding: 80px;
`;

export const StyledStoryCard = styled.div`
  flex: 1;
  border: 1px solid ${({ theme }) => theme.BACKGROUND_MODIFIER_ACCENT};
  background: rgb(9, 4, 66);
  background: linear-gradient(
    138deg,
    rgba(16, 8, 80, 0.8870141806722689) 0%,
    rgba(30, 16, 94, 0.40802258403361347) 33%,
    rgba(22, 12, 82, 0.6012998949579832) 100%
  );
  border-radius: 6px;
  margin: 2rem;
  position: relative;
  width: 60%;

  @media only screen and (max-width: 768px) {
    width: 90%;
  }
`;

export const StyledHeroArea = styled.section`
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 50vh;
  width: 100%;

  &:before {
    content: "";
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background-image: url(${img.src});
    background-position: center;
    background-size: cover;
    background-repeat: no-repeat;
    background-attachment: fixed;
    opacity: 50%;
  }
`;
