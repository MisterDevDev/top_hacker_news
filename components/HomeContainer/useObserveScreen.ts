import { IStory } from "@/typings";
import { useState, useMemo, useEffect } from "react";

export default function useObserveScreen(options: any, targetRef: any) {
  const [isVisible, setIsVisible] = useState();

  const callbackFunction = (stories: any) => {
    const [story] = stories;
    setIsVisible(story.isIntersecting);
  };

  const optionsMemo = useMemo(() => {
    return options;
  }, [options]);

  useEffect(() => {
    const observer = new IntersectionObserver(callbackFunction, optionsMemo);
    const currentTarget = targetRef.current;
    if (currentTarget) observer.observe(currentTarget);

    return () => {
      if (currentTarget) observer.unobserve(currentTarget);
    };
  }, [targetRef, optionsMemo]);

  return isVisible;
}
