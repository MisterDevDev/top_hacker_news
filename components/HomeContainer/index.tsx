import { IStory } from "@/typings";
import * as Styles from "./styles";
import { StoryCard } from "../StoryCard";
import { HeroParticles } from "./Particles";
import { useState, useRef, useEffect } from "react";
import useObserveScreen from "./useObserveScreen";

interface Props {
  topStories: IStory[];
}

export const HomeContainer = (props: Props) => {
  const [visibleStories, setVisibleStories] = useState(10);
  const lastStory = useRef(null);

  const isVisible = useObserveScreen(
    {
      root: null,
      rootMargin: "0px",
      threshold: 0.3,
    },
    lastStory
  );

  useEffect(() => {
    if (isVisible) addVisibleStories();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isVisible]);

  return (
    <main>
      {renderTitleSection()}
      {renderStoryContainer()}
    </main>
  );

  function addVisibleStories() {
    if (visibleStories <= 90) setVisibleStories(visibleStories + 10);
  }

  function renderTitleSection() {
    return (
      <>
        <HeroParticles />
        <Styles.StyledHeroArea>
          <Styles.StyledSubtitleBox>
            <Styles.StyledTitle>Hacker News</Styles.StyledTitle>
            <Styles.StyledGradientText>
              <i>Top 100</i>
            </Styles.StyledGradientText>
          </Styles.StyledSubtitleBox>
        </Styles.StyledHeroArea>
      </>
    );
  }

  function renderStoryContainer() {
    return <Styles.StyledStoryContainer>{renderStories()}</Styles.StyledStoryContainer>;
  }

  function renderStories() {
    return (
      <>
        {props.topStories.map((story: IStory, idx) => {
          if (story.deleted) return <></>;
          if (idx + 1 === visibleStories) {
            return (
              <Styles.StyledStoryCard key={story.id}>
                <div ref={lastStory} aria-label="news-story">
                  <StoryCard story={story} />
                </div>
              </Styles.StyledStoryCard>
            );
          }
          if (idx + 1 < visibleStories) {
            return (
              <Styles.StyledStoryCard key={story.id}>
                <div aria-label="news-story">
                  <StoryCard story={story} />
                </div>
              </Styles.StyledStoryCard>
            );
          }
        })}
      </>
    );
  }
};
