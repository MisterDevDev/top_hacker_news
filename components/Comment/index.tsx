import { IComment } from "@/typings";
import { useState, useEffect } from "react";
import HTMLparse from "html-react-parser";
import * as Styles from "./styles";
import dayjs from "dayjs";
import relativeTime from "dayjs/plugin/relativeTime";
import { CommentsContainer } from "../CommentsContainer";
import { IoIosReturnRight } from "react-icons/io";
import { CgCornerUpRight } from "react-icons/cg";

dayjs.extend(relativeTime);

interface Props {
  kid_id: number;
  isNestedReply: boolean;
}

export const Comment = (props: Props) => {
  const [comment, setComment] = useState<IComment | undefined>();
  const [showReplies, setShowReplies] = useState(false);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    (async () => {
      setLoading(true);
      const request = `https://hacker-news.firebaseio.com/v0/item/${props.kid_id}.json?print=pretty`;
      const response = await fetch(request);
      setComment(await response.json());
      setLoading(false);
    })();
  }, [props.kid_id]);

  return <>{renderComment()}</>;

  function renderComment() {
    if (comment?.deleted) return <>{renderDeletedComment()}</>;
    if (loading) return <>{renderLoading()}</>;
    return (
      <>
        <Styles.StyledCommentContainer loading={loading ? 1 : 0}>
          {renderAuthor()}
          {renderText()}
        </Styles.StyledCommentContainer>
        {renderReplies()}
      </>
    );
  }

  function renderAuthor() {
    return (
      <Styles.StyledAuthorContainer>
        <Styles.StyledAuthorSpan>{comment?.by ?? ""}</Styles.StyledAuthorSpan> {" " + getTime()}
      </Styles.StyledAuthorContainer>
    );
  }

  function renderText() {
    return <Styles.StyledParsedSpan>{HTMLparse(comment?.text ?? "")}</Styles.StyledParsedSpan>;
  }

  function getTime() {
    if (Number(comment?.time ?? 0)) {
      return dayjs.unix(comment?.time ?? 0).fromNow();
    }
    return "";
  }

  function renderReplies() {
    if (comment?.kids?.[0])
      return (
        <>
          {!showReplies && (
            <Styles.StyledReplyLink>
              <IoIosReturnRight size={18} />
              <Styles.StyledHoverSpan onClick={() => setShowReplies(true)}>
                {comment.kids.length + getReply()}
              </Styles.StyledHoverSpan>
            </Styles.StyledReplyLink>
          )}
          {showReplies && !props.isNestedReply && (
            <Styles.StyledReplyLink>
              <CgCornerUpRight size={18} />
              <Styles.StyledHoverSpan onClick={() => setShowReplies(false)}>
                {"Close " + getReply()}
              </Styles.StyledHoverSpan>
            </Styles.StyledReplyLink>
          )}
          {showReplies && (
            <Styles.StyledReplyContainer>
              <CommentsContainer isReply kids={comment.kids} />
            </Styles.StyledReplyContainer>
          )}
        </>
      );
    return <></>;
  }

  function getReply() {
    return (comment?.kids?.length ?? 0) > 1 ? " Replies" : " Reply";
  }

  function renderDeletedComment() {
    <Styles.StyledCommentContainer loading={loading ? 1 : 0}>
      <span>
        <i>Post Deleted</i>
      </span>
    </Styles.StyledCommentContainer>;
  }

  function renderLoading() {
    <Styles.StyledCommentContainer loading={loading ? 1 : 0}>
      <span>Loading...</span>
    </Styles.StyledCommentContainer>;
  }
};
