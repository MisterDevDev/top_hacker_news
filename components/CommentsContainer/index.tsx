import { useState } from "react";
import { Comment } from "../Comment";
import * as Styles from "./styles";
import { RxArrowUp } from "react-icons/rx";

interface Props {
  kids: number[];
  isReply?: boolean;
}

export const CommentsContainer = (props: Props) => {
  const [visibleComments, setVisibleComments] = useState(0);

  return <>{renderCommentsContainer()}</>;

  function renderCommentsContainer() {
    return (
      <Styles.StyledContentContainer>
        {props.kids.map((kid_id: number, idx) => {
          if (idx <= visibleComments)
            return <Comment key={kid_id} kid_id={kid_id} isNestedReply={!!props.isReply} />;
        })}
        {visibleComments < props.kids.length - 1 && renderMoreLink()}
      </Styles.StyledContentContainer>
    );
  }

  function renderMoreLink() {
    if (props.isReply) return renderMoreReplies();
    return renderMoreComments();
  }

  function renderMoreReplies() {
    return (
      <Styles.StyledHoverSpan
        style={{ paddingLeft: "20px" }}
        onClick={() => setVisibleComments(visibleComments + 3)}
      >
        <RxArrowUp size={12} /> More Replies
      </Styles.StyledHoverSpan>
    );
  }

  function renderMoreComments() {
    return (
      <Styles.StyledMoreComments style={{ textAlign: "center" }}>
        <Styles.StyledHoverSpan
          style={{ paddingLeft: "10px" }}
          onClick={() => setVisibleComments(visibleComments + 3)}
        >
          More Comments
        </Styles.StyledHoverSpan>
      </Styles.StyledMoreComments>
    );
  }
};
