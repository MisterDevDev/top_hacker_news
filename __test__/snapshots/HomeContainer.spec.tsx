import React from "react";
import "@testing-library/jest-dom";
import { HomeContainer } from "@/components/HomeContainer";
import { mockIntersectionObserver } from "jsdom-testing-mocks";
import { readFakeData } from "../mock_data";
import renderer from "react-test-renderer";

mockIntersectionObserver();

it("Should render correctly", async () => {
  const { fakeStories } = await readFakeData();
  const tree = renderer.create(<HomeContainer topStories={fakeStories} />);
  expect(tree.toJSON()).toMatchSnapshot();
  tree.unmount();
});
