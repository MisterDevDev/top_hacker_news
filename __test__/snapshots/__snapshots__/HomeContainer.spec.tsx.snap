// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<main>
  <div
    id="tsparticles"
  >
    <canvas
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </div>
  <section
    className="sc-jSUZER hnBDmO"
  >
    <div
      className="sc-hLBbgP iqoPhb"
    >
      <h1
        className="sc-gswNZR cczJop"
      >
        Hacker News
      </h1>
      <h4
        className="sc-dkrFOg dkUkXh"
      >
        <i>
          Top 100
        </i>
      </h4>
    </div>
  </section>
  <div
    className="sc-bcXHqe fufuNP"
  >
    <div
      className="sc-eDvSVe erSdmQ"
    >
      <div
        aria-label="news-story"
      >
        <div
          className="sc-ipEyDJ KSSCD"
        >
          <div
            className="sc-kDvujY faROvf"
          >
            <div
              className="sc-jrcTuL"
            >
              1666 points by 
              <span
                className="sc-ksBlkl lbYNbX"
              >
                fileeditview
              </span>
               
              <br />
              (kentnerburn.com)
            </div>
          </div>
          <div
            className="sc-bqWxrE gpQyvZ"
          />
          <h2
            className="sc-iBYQkv iijhMY"
            onClick={[Function]}
          >
            <span
              className="sc-ftTHYK eunGkG"
            >
              A cab ride I'll never forget (1999)
            </span>
          </h2>
          <div
            className="sc-csuSiG fwVXDo"
          >
            <span
              className="sc-ftTHYK eunGkG"
              onClick={[Function]}
            >
              View 3 Comments
            </span>
            <span
              aria-label="posted-time"
            >
              Jan 16, 2023
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-eDvSVe erSdmQ"
    >
      <div
        aria-label="news-story"
      >
        <div
          className="sc-ipEyDJ KSSCD"
        >
          <div
            className="sc-kDvujY faROvf"
          >
            <div
              className="sc-jrcTuL"
            >
              1297 points by 
              <span
                className="sc-ksBlkl lbYNbX"
              >
                colesantiago
              </span>
               
              <br />
              (blog.google)
            </div>
          </div>
          <div
            className="sc-bqWxrE gpQyvZ"
          />
          <h2
            className="sc-iBYQkv iijhMY"
            onClick={[Function]}
          >
            <span
              className="sc-ftTHYK eunGkG"
            >
              Google to reduce workforce by 12k
            </span>
          </h2>
          <div
            className="sc-csuSiG fwVXDo"
          >
            <span
              className="sc-ftTHYK eunGkG"
              onClick={[Function]}
            >
              View 1349 Comments
            </span>
            <span
              aria-label="posted-time"
            >
              Jan 20, 2023
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-eDvSVe erSdmQ"
    >
      <div
        aria-label="news-story"
      >
        <div
          className="sc-ipEyDJ KSSCD"
        >
          <div
            className="sc-kDvujY faROvf"
          >
            <div
              className="sc-jrcTuL"
            >
              1071 points by 
              <span
                className="sc-ksBlkl lbYNbX"
              >
                georgehill
              </span>
               
              <br />
              (youtube.com)
            </div>
          </div>
          <div
            className="sc-bqWxrE gpQyvZ"
          />
          <h2
            className="sc-iBYQkv iijhMY"
            onClick={[Function]}
          >
            <span
              className="sc-ftTHYK eunGkG"
            >
              Let's build GPT: from scratch, in code, spelled out by Andrej Karpathy [video]
            </span>
          </h2>
          <div
            className="sc-csuSiG fwVXDo"
          >
            <span
              className="sc-ftTHYK eunGkG"
              onClick={[Function]}
            >
              View 101 Comments
            </span>
            <span
              aria-label="posted-time"
            >
              Jan 17, 2023
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-eDvSVe erSdmQ"
    >
      <div
        aria-label="news-story"
      >
        <div
          className="sc-ipEyDJ KSSCD"
        >
          <div
            className="sc-kDvujY faROvf"
          >
            <div
              className="sc-jrcTuL"
            >
              885 points by 
              <span
                className="sc-ksBlkl lbYNbX"
              >
                half-kh-hacker
              </span>
               
              <br />
              (maia.crimew.gay)
            </div>
          </div>
          <div
            className="sc-bqWxrE gpQyvZ"
          />
          <h2
            className="sc-iBYQkv iijhMY"
            onClick={[Function]}
          >
            <span
              className="sc-ftTHYK eunGkG"
            >
              How to own an airline in 3 easy steps and grab the TSA nofly list along the way
            </span>
          </h2>
          <div
            className="sc-csuSiG fwVXDo"
          >
            <span
              className="sc-ftTHYK eunGkG"
              onClick={[Function]}
            >
              View 547 Comments
            </span>
            <span
              aria-label="posted-time"
            >
              Jan 19, 2023
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-eDvSVe erSdmQ"
    >
      <div
        aria-label="news-story"
      >
        <div
          className="sc-ipEyDJ KSSCD"
        >
          <div
            className="sc-kDvujY faROvf"
          >
            <div
              className="sc-jrcTuL"
            >
              871 points by 
              <span
                className="sc-ksBlkl lbYNbX"
              >
                ColinWright
              </span>
               
              <br />
              (somethingorotherwhatever.com)
            </div>
          </div>
          <div
            className="sc-bqWxrE gpQyvZ"
          />
          <h2
            className="sc-iBYQkv iijhMY"
            onClick={[Function]}
          >
            <span
              className="sc-ftTHYK eunGkG"
            >
              Wobbly clock
            </span>
          </h2>
          <div
            className="sc-csuSiG fwVXDo"
          >
            <span
              className="sc-ftTHYK eunGkG"
              onClick={[Function]}
            >
              View 195 Comments
            </span>
            <span
              aria-label="posted-time"
            >
              Jan 16, 2023
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-eDvSVe erSdmQ"
    >
      <div
        aria-label="news-story"
      >
        <div
          className="sc-ipEyDJ KSSCD"
        >
          <div
            className="sc-kDvujY faROvf"
          >
            <div
              className="sc-jrcTuL"
            >
              861 points by 
              <span
                className="sc-ksBlkl lbYNbX"
              >
                ValentineC
              </span>
               
              <br />
              (apple.com)
            </div>
          </div>
          <div
            className="sc-bqWxrE gpQyvZ"
          />
          <h2
            className="sc-iBYQkv iijhMY"
            onClick={[Function]}
          >
            <span
              className="sc-ftTHYK eunGkG"
            >
              MacBook Pro featuring M2 Pro and M2 Max
            </span>
          </h2>
          <div
            className="sc-csuSiG fwVXDo"
          >
            <span
              className="sc-ftTHYK eunGkG"
              onClick={[Function]}
            >
              View 1025 Comments
            </span>
            <span
              aria-label="posted-time"
            >
              Jan 17, 2023
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-eDvSVe erSdmQ"
    >
      <div
        aria-label="news-story"
      >
        <div
          className="sc-ipEyDJ KSSCD"
        >
          <div
            className="sc-kDvujY faROvf"
          >
            <div
              className="sc-jrcTuL"
            >
              860 points by 
              <span
                className="sc-ksBlkl lbYNbX"
              >
                subharmonicon
              </span>
               
              <br />
              (smithsonianmag.com)
            </div>
          </div>
          <div
            className="sc-bqWxrE gpQyvZ"
          />
          <h2
            className="sc-iBYQkv iijhMY"
            onClick={[Function]}
          >
            <span
              className="sc-ftTHYK eunGkG"
            >
              Photos capture life inside a drop of seawater
            </span>
          </h2>
          <div
            className="sc-csuSiG fwVXDo"
          >
            <span
              className="sc-ftTHYK eunGkG"
              onClick={[Function]}
            >
              View 187 Comments
            </span>
            <span
              aria-label="posted-time"
            >
              Jan 19, 2023
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-eDvSVe erSdmQ"
    >
      <div
        aria-label="news-story"
      >
        <div
          className="sc-ipEyDJ KSSCD"
        >
          <div
            className="sc-kDvujY faROvf"
          >
            <div
              className="sc-jrcTuL"
            >
              841 points by 
              <span
                className="sc-ksBlkl lbYNbX"
              >
                blegh
              </span>
               
              <br />
              (milk.com)
            </div>
          </div>
          <div
            className="sc-bqWxrE gpQyvZ"
          />
          <h2
            className="sc-iBYQkv iijhMY"
            onClick={[Function]}
          >
            <span
              className="sc-ftTHYK eunGkG"
            >
              What not to write on your security clearance form (1988)
            </span>
          </h2>
          <div
            className="sc-csuSiG fwVXDo"
          >
            <span
              className="sc-ftTHYK eunGkG"
              onClick={[Function]}
            >
              View 531 Comments
            </span>
            <span
              aria-label="posted-time"
            >
              Jan 19, 2023
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-eDvSVe erSdmQ"
    >
      <div
        aria-label="news-story"
      >
        <div
          className="sc-ipEyDJ KSSCD"
        >
          <div
            className="sc-kDvujY faROvf"
          >
            <div
              className="sc-jrcTuL"
            >
              708 points by 
              <span
                className="sc-ksBlkl lbYNbX"
              >
                chazeon
              </span>
               
              <br />
              (furbo.org)
            </div>
          </div>
          <div
            className="sc-bqWxrE gpQyvZ"
          />
          <h2
            className="sc-iBYQkv iijhMY"
            onClick={[Function]}
          >
            <span
              className="sc-ftTHYK eunGkG"
            >
              The Shit Show
            </span>
          </h2>
          <div
            className="sc-csuSiG fwVXDo"
          >
            <span
              className="sc-ftTHYK eunGkG"
              onClick={[Function]}
            >
              View 315 Comments
            </span>
            <span
              aria-label="posted-time"
            >
              Jan 15, 2023
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-eDvSVe erSdmQ"
    >
      <div
        aria-label="news-story"
      >
        <div
          className="sc-ipEyDJ KSSCD"
        >
          <div
            className="sc-kDvujY faROvf"
          >
            <div
              className="sc-jrcTuL"
            >
              647 points by 
              <span
                className="sc-ksBlkl lbYNbX"
              >
                georgehill
              </span>
               
              <br />
              (reuters.com)
            </div>
          </div>
          <div
            className="sc-bqWxrE gpQyvZ"
          />
          <h2
            className="sc-iBYQkv iijhMY"
            onClick={[Function]}
          >
            <span
              className="sc-ftTHYK eunGkG"
            >
              Microsoft to lay off 11k employees
            </span>
          </h2>
          <div
            className="sc-csuSiG fwVXDo"
          >
            <span
              className="sc-ftTHYK eunGkG"
              onClick={[Function]}
            >
              View 453 Comments
            </span>
            <span
              aria-label="posted-time"
            >
              Jan 17, 2023
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
